### DISCUSSION TITLE
npm question

### DISCUSSION THREAD

**Discussion by krauthaus** [nextcloud/all-in-one#1131] [Questions] [answered]
Hi there! Since days I am trying to get this reverse proxy running, but till now I got no luck since I seem to miss some important detail in the documentation.

I forwarded all ports (80.443.8443.8080.3478) on my unifi router. I also did a local dns entry for the host IP with my xxx.mydomain.xx.
I installed npm with nextcloud aio (the docker compose file from the docs.), and It does forward port 80, but not 443. To forward https it looks like I need to get the cert on npm. (here I can get a cert from npm for my subdomain)

An here I am stuck. Now i got a valid letsencrypt cert for xxx.mydomain.com, and I forwarding ist to my host server 192.168.0.2:11000. (example ip)

But on the mastercontainer-backend I cant enter the domain since it tells me the reverse proxy is not properly configured. Skipping domain validation does not help.

Should I copy the npm-generated cert now to the mastercontainer/data/custom_ssl folder? Or should I somewhere change the path to the npm-cert folder? 

Cant see 502s any more xD

**Comment by Zoey2936** [comment, marked as answer]
Can you send your npm configuration? [(Solution: Scheme was set to https, but it needs to be http)](https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#nginx-proxy-manager)

**Reply by krauthaus** [reply]
`version: "3.8"

volumes:
nextcloud_aio_mastercontainer:
name: nextcloud_aio_mastercontainer
services:
nginx_proxy_manager:
image: jc21/nginx-proxy-manager:latest
restart: unless-stopped
container_name: nginx-proxy-manager
network_mode: host
#environment: # Uncomment this if IPv6 is not enabled on your host
#- DISABLE_IPV6=true # Uncomment this if IPv6 is not enabled on your host
volumes:
- ./data:/data
- ./letsencrypt:/etc/letsencrypt
nextcloud:
image: nextcloud/all-in-one:latest
restart: unless-stopped
container_name: nextcloud-aio-mastercontainer
depends_on:
- nginx_proxy_manager
volumes:
- nextcloud_aio_mastercontainer:/mnt/docker-aio-config
- /var/run/docker.sock:/var/run/docker.sock:ro
ports:
- 8080:8080
environment:
- APACHE_PORT=11000
- APACHE_DISABLE_REWRITE_IP=1
- NEXTCLOUD_TRUSTED_DOMAINS=xxx.xxx.at 192.168.0.2 # Your domain name + proxy host IP
- TRUSTED_PROXIES=192.168.0.2 # proxy host IP
`

**Reply by Zoey2936** [reply]
Oh, I meant a screenshot from the Nextcloud configuration of npms web ui

**Reply by Zoey2936** [reply]
Simple error, you need to set your scheme to http, not https

**Reply by krauthaus** [reply]
Damn, I was sure I tried this. Thank you very much!
